tests/sfko/sfko/report.py

```diff
@@ -0,0 +1,151 @@
+    # next_run = datetime.datetime.combine(then.date(), _RESULTS_SCHED_TIME)
+    _log.debug('Next scheduler run at %s'%(next_run))
+    return (next_run - now).total_seconds()
+
+def init_scheduler():
+    _log.debug('Initializing scheduler')
+    _RESULTS_SCHEDULER.enter(_get_next_schedule(), 0, collect_results)
+    _RESULTS_SCHEDULER.start()
+
+
+# HIPCHAT_URL = 'https://scalitysocial.hipchat.com/v2/room/4661513/notification'
+HIPCHAT_URL = 'https://scalitysocial.hipchat.com/v2/room/%s/notification'%config.results.hipchat.room
+def notify(msg, color = 'yellow', notify = False):
+    data = dict(
+        message = msg,
+        message_format = 'text',
+        notify = notify,
+        color = color
+    )
+    token = dict(auth_token=config.results.hipchat.token)
+    resp = requests.post(HIPCHAT_URL, json = data, params=token)
+    _log.debug(resp)
+
+
+x = '''Op Failed:
+  backend:\t{backend}
+  bucket:\t{bucket}
+  objects:\t{obj_count}
+  size:\t\t{obj_size}
+  versioning:\t{versioning}
+  replication:\t{replication}
+  transient:\t\t{transient}
+  expiration:\t{expiration}'''
+
+
+FAILURE_DETAILS = [
+    'backend',
+    'bucket',
+    'objects',
+    'size',
+    'versioning',
+    'replication',
+    'transient',
+    'expiration'
+]
+
+def sizeof_fmt(num, suffix='B'):
+    for unit in ['','Ki','Mi','Gi','Ti','Pi','Ei','Zi']:
+        if abs(num) < 1024.0:
+            return "%3.1f%s%s" % (num, unit, suffix)
+        num /= 1024.0
+    return "%.1f%s%s" % (num, 'Yi', suffix)
+
+def build_failures(op, bucket, oproxy, op_type = 'check'):
+    if bucket.replication is not False:
+        repl = bucket.replication.backend.type.friendly()
+    else:
+        repl = False
+    deets = OrderedDict(
+        backend='%s:%s'%(bucket.backend.type.friendly(), bucket.backend.bucket),
+        bucket=bucket.name,
+        objects=oproxy.count,
+        size=sizeof_fmt(oproxy.filesize),
+        versioning=bucket.versioning,
+        replication=repl,
+        transient=bucket.transient,
+        expiration=bucket.expiration
+    )
+    lines = ['   Failed %s: %s'%(op_type, op.name)]
+    for key in FAILURE_DETAILS:
+        if deets[key] is not None and deets[key] is not False:
+            lines.append('%s: %s'%(key, deets[key]))
+    return '\n      '.join(lines)
+
+def push_to_chat(results):
+    lines = []
+    failed = 0
+    for ts, (scenario, success, failed_tests, failed_checks) in results.items():
+        status = '(successful)' if success else '(failed)'
+        sline = REPORT_LINE.format(scenario=scenario.name, ts=ts, status=status)
+        lines.append(sline)
+        if not success:
+            for op, (bucket, oproxy) in chain(failed_tests, failed_checks):
+                op_type = 'test' if (op, (bucket, oproxy)) in failed_tests else 'check'
+                lines.append(build_failures(op, bucket, oproxy, op_type))
+                failed += 1
+    if failed == 0:
+        color = 'green'
+    elif failed/len(results) < 0.5:
```

Comment:
This is a good attempt at giving a colorful clue of how bad the test is, but apart from green or red I'm not sure yellow would bring something without a meaningful metric, so maybe skip this unless we have a clear definition of what is a "partially successful" test?